### PR TITLE
Enable datacatalog client to use admin's auth option

### PR DESCRIPTION
**Pull request by mayitbeegh**
# TL;DR
Enable datacatalog client to use admin's auth option. See https://github.com/flyteorg/flyteidl/pull/252
TODO: fix flyteidl version

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Review by EngHabu — COMMENTED** [review]
Awesome! Once we release flyteidl, this can go in
